Sources/VitaminUIKit/Components/Button/VitaminButton.swift

```diff
@@ -275,17 +258,135 @@ extension VitaminButton.Size {
         .button
     }
 
-    var horizontalInset: CGFloat {
+    func horizontalInset(iconType: VitaminButton.IconType) -> CGFloat {
+        if case .alone = iconType {
+            return 12
+        }
+
         switch self {
         case .medium: return 20
         case .large: return 40
         }
     }
 
-    var verticalInset: CGFloat {
+    func verticalInset(iconType: VitaminButton.IconType) -> CGFloat {
+        if case .alone = iconType {
+            return 12
+        }
+
         switch self {
         case .medium: return 16
         case .large: return 20
         }
     }
+
+    func defaultIconSize(iconType: VitaminButton.IconType) -> CGFloat {
+        if case .alone = iconType {
+            switch self {
+            case .medium : return 24
+            case .large: return 32
+            }
+        } else {
+            switch self {
+            case .medium : return 20
+            case .large: return 24
+            }
+        }
+    }
+}
+
+// - MARK: Icon managemant
+
+extension VitaminButton {
+    public func setIconType(_ iconType: IconType, for state: UIControl.State) {
+        iconTypes[state] = iconType
+        applyIcon(for: state)
+    }
+
+    public func getIconType(for state: UIControl.State) -> IconType {
+        guard let iconType = iconTypes[state] else {
+            return iconTypes[.normal] ?? .none
+        }
+        return iconType
+    }
+
+    private func applyIcon(for state: UIControl.State) {
+        self.invalidateIntrinsicContentSize()
+        let iconTypeForState = getIconType(for: state)
+        switch iconTypeForState {
+        case .none:
+            break
+        case let .trailing(image, renderingMode):
+            self.commonApplyIcon(
+                image: image,
+                iconType: iconTypeForState,
+                state: state,
+                renderingMode: renderingMode)
+        case let .leading(image, renderingMode):
+            self.commonApplyIcon(
+                image: image,
+                iconType: iconTypeForState,
+                state: state,
+                renderingMode: renderingMode)
+        case let .alone(image, renderingMode):
+            self.setTitle("", for: state)
```

Comment:
Maybe use `nil` for this.
```suggestion
            self.setTitle(nil, for: state)
```